numpy/_utils/__init__.py

```diff
@@ -27,3 +29,59 @@ def decorator(func):
             func.__module__ = module
         return func
     return decorator
+
+
+def _rename_parameter(old_names, new_names, dep_version=None):
+    """
+    Generate decorator for backward-compatible keyword renaming.
+
+    Apply the decorator generated by `_rename_parameter` to functions with a
+    renamed parameter to maintain backward-compatibility.
+
+    After decoration, the function behaves as follows:
+    If only the new parameter is passed into the function, behave as usual.
+    If only the old parameter is passed into the function (as a keyword), raise
+    a DeprecationWarning if `dep_version` is provided, and behave as usual
+    otherwise.
+    If both old and new parameters are passed into the function, raise a
+    DeprecationWarning if `dep_version` is provided, and raise the appropriate
+    TypeError (function got multiple values for argument).
+
+    Parameters
+    ----------
+    old_names : list of str
+        Old names of parameters
+    new_name : list of str
+        New names of parameters
+    dep_version : str, optional
+        Version of NumPy in which old parameter was deprecated in the format
+        'X.Y.Z'. If supplied, the deprecation message will indicate that
+        support for the old parameter will be removed in version 'X.Y+2.Z'
```

Comment:
Deprecations in NumPy are for two minor versions (like SciPy), or is the schedule different?

Reply:
It's the same indeed.

Reply:
@rgommers do you think I should email mailing list about this change or wait for a better time?

Reply:
Probably first get the review completed I'd say, since the end result may still change. It's a pretty minor deprecation, since I'd expect usage of the keyword form to be rare.